sql/plan/process.go

```diff
@@ -314,3 +390,26 @@ func partitionName(p sql.Partition) string {
 	}
 	return string(p.Key())
 }
+
+// IsDdlNode returns whether the node given is a DDL operation, which includes things like SHOW commands. In general,
+// these are nodes that interact only with schema and the catalog, not with any table rows.
+func IsDdlNode(node sql.Node) bool {
+	switch node.(type) {
+	case *CreateTable, *DropTable, *Truncate,
+		*AddColumn, *ModifyColumn, *DropColumn,
+		*CreateDB, *DropDB,
+		*RenameTable, *RenameColumn,
+		*CreateIndex, *AlterIndex, *DropIndex,
+		*CreateForeignKey, *DropForeignKey,
+		*CreateTrigger, *DropTrigger,
+		*ShowTables, *ShowCreateTable,
+		*ShowTriggers, *ShowCreateTrigger,
+		*ShowDatabases, *ShowCreateDatabase,
+		*ShowColumns, *ShowIndexes,
+		*ShowProcessList, *ShowTableStatus,
+		*ShowVariables, *ShowWarnings:
```

Comment:
Need to add `*CreateProcedure` and `*DropProcedure` here. That's my fault. I would prefer an interface overall over this since it's easy(er) to forget.

Reply:
Good catch, probably need to have some kind of marker interface for the various kinds of statements at some point.